mqtt/mqtt-bridge/src/persist/memory/loader.rs

```diff
@@ -18,7 +18,7 @@ pub struct InMemoryMessageLoader {
 }
 
 impl InMemoryMessageLoader {
-    pub async fn new(state: Arc<Mutex<WakingMap>>, batch_size: usize) -> Self {
+    pub fn new(state: &Arc<Mutex<WakingMap>>, batch_size: usize) -> Self {
```

Comment:
I think you can still consume `state` and omit cloning later

Reply:
I made these changes to the loader to be able to integrate with persistence, but Andrew's PR completely changes the API so all changes in persist folder are throw-away